File: src/main/java/lesson3/Human.java
package lesson3;

class Human implements Participiant {
    static int maxJumpHeight = 10;
    static int maxRunDistance = 20;

    @Override
    public boolean jump(int height) {
        if (height <= Human.maxJumpHeight) {
            System.out.println("Человек успешно прыгнул");
            return true;
        }
        System.out.println("Человек не прыгнул");
        return false;
    }

    @Override
    public boolean run(int distance) {
        if (distance <= Human.maxRunDistance) {
            System.out.println("Человек успешно пробежал");
            return true;
        }
        System.out.println("Человек не пробежал");
        return false;
    }
}

File: src/main/kotlin/lesson3_kt/Human.kt
package lesson3_kt

class Human : Participian{
    companion object {
        val HUMAN_MAX_DISTANCE = 10
        val HUMAN_MAX_HEIGHT = 20
    }
    override fun run(distance: Int) : Boolean = performAction(
        value = distance,
        maxValue = HUMAN_MAX_DISTANCE,
        action = "пробежал",
        actor = "человек"
    )

    override fun jump(height: Int) : Boolean= performAction(
        value = height,
        maxValue= HUMAN_MAX_HEIGHT,
        action = "прыгнул",
        actor = "человек"
    )
}